### ISSUE TITLE
Notifications from shared-with-me grains with missing icons don't display correctly

### ISSUE THREAD

**Issue by kentonv** [sandstorm-io/sandstorm#2633]
When testing @dwrensha Wekan changes, I observed that the bell menu was showing up empty even when I definitely had Notifications waiting. The console showed the stack trace below. I suspect this has to do with the fact that this Wekan build is missing icons.

Note that the problem only occurs for users who do not own the grain. The grain owner sees notifications rendered correctly with an identicon.

```
Exception in template helper: TypeError: Cannot read property 'assetId2xDpi' of null
at e (https://oasis.sandstorm.io/ad7921a02c5a97967428a0c554f60da892aad1d5.js?meteor_js_resource=true:146:2999)
at Function.g.iconSrcForDenormalizedGrainMetadata (https://oasis.sandstorm.io/ad7921a02c5a97967428a0c554f60da892aad1d5.js?meteor_js_resource=true:146:3416)
at https://oasis.sandstorm.io/ad7921a02c5a97967428a0c554f60da892aad1d5.js?meteor_js_resource=true:196:13560
at https://oasis.sandstorm.io/ad7921a02c5a97967428a0c554f60da892aad1d5.js?meteor_js_resource=true:42:1836
at https://oasis.sandstorm.io/ad7921a02c5a97967428a0c554f60da892aad1d5.js?meteor_js_resource=true:42:1659
at Array.forEach (native)
at Function.A.each.A.forEach (https://oasis.sandstorm.io/ad7921a02c5a97967428a0c554f60da892aad1d5.js?meteor_js_resource=true:1:865)
at h.Cursor.forEach (https://oasis.sandstorm.io/ad7921a02c5a97967428a0c554f60da892aad1d5.js?meteor_js_resource=true:42:1581)
at h.Cursor.map (https://oasis.sandstorm.io/ad7921a02c5a97967428a0c554f60da892aad1d5.js?meteor_js_resource=true:42:1805)
at e (https://oasis.sandstorm.io/ad7921a02c5a97967428a0c554f60da892aad1d5.js?meteor_js_resource=true:196:12601)
```

**Comment by zarvox** [comment]
Looks like `iconObj` in [iconSrcFor](https://github.com/sandstorm-io/sandstorm/blob/8d5092309068eb892bc0233f5a1b22ed70bcc07b/shell/packages/sandstorm-identicons/helpers.js#L75) was `null` rather than `undefined`.

Tracing back, this object came from https://github.com/sandstorm-io/sandstorm/blob/8d5092309068eb892bc0233f5a1b22ed70bcc07b/shell/packages/sandstorm-db/db.js#L1258 which suggests that icons.appGrid had the value `null`.

Is that what you're supposed to get when node-capnp decodes a message that has that field unset, or do we expect to get `undefined` in that context?  I *think* I'd expect to get an object `{ unknown: null }` based on https://github.com/sandstorm-io/sandstorm/blob/8d5092309068eb892bc0233f5a1b22ed70bcc07b/src/sandstorm/package.capnp#L276 if the Icon were set.

**Comment by kentonv** [comment]
If `appGrid` was null in the capnp then it should have been omitted from the JSON conversion, so should come up undefined.

There seem to be 22 ApiTokens total on Oasis which have denormalizedGrainMetadata.icon = null...

**Comment by jparyani** [comment]
I am unable to reproduce this locally with devAccounts.

**Comment by dwrensha** [comment]
I am able to reproduce the problem. See #2764. Probably not worth a migration.

**Comment by dwrensha** [comment]
Fixed by #2758.
